cornflow/schemas/schema_manager.py

```
@@ -0,0 +1,368 @@
+        """
+        Create a draft jsonschema from a json file of data.
+        
+        :param path: path to the json file.
+        :param save_path: path where to save the generated schema.
+        
+        :return: the generated schema.
+        """
+        
+        file = self.load_json(path)
+        
+        builder = SchemaBuilder()
+        builder.add_schema({"type": "object", "properties": {}})
+        builder.add_object(file)
+        
+        self.draft_schema = builder.to_json()
+        if save_path is not None:
+            self.save_json(self.draft_schema, save_path)
+        return self.draft_schema
+    
+    """
+    Functions used to translate jsonschema to schema_dict
+    """
+    def _create_data_schema(self, item):
+        """
+        Add a schema to self.schema_dict["DataSchema"]
+
+        :param item: (key, value) of a dict. The key cotain the name of the schema and the value containt its content.
+
+        return the schema dict.
+        """
+    
+        name = item[0]
+        content = item[1]
+    
+        schema = [dict(name=name,
+                      type=self._get_schema_name(name),
+                      many=("type" in content and content["type"] == "array"),
+                      required=True)]
+        self.schema_dict["DataSchema"] +=schema
+    
+        return schema
+    
+    def _get_element_dict(self, item, required_list=None):
+        """
+        Parse an item (key, value) from the jsonschema and return the corresponding dict.
+        
+        :param item: An item from the jsonschema (key, value)
+        :param required_list: A list of names corresponding to the required fields in the parent object
+        
+        :return A dict element for a schema_dict.
+        """
+        if required_list is None:
+            required_list = []
+        
+        name = item[0]
+        content = item[1]
+        if "type" not in content:
+            if "$ref" in content:
```

what happens when this inside if is false? do we get an error?

---

good question, I do not know what we should do if no type is given:
- raise an error here?
- make it a string by default?
- raise an error when transforming it into a flask schema?

---

I have no idea what the code does actually, I just noticed the ifs. If it's wrong (unexpected): we should raise an error saying "incorrect jsonschema" and that the object `name` is missing a type. I've just checked that "raise TypeError()" exists so you could use that haha.